crates/primitives/starknet/src/transaction/mod.rs

```diff
@@ -94,7 +95,56 @@ impl Transaction {
             sender_address: StarknetContractAddress::try_from(StarkFelt::new(self.sender_address).unwrap()).unwrap(),
             calldata: self.call_entrypoint.to_starknet_call_entry_point().calldata,
             entry_point_selector: None,
-        })
+        }
+    }
+
+    /// Converts a transaction to a blockifier declare transaction
+    ///
+    /// # Arguments
+    ///
+    /// * `self` - The transaction to convert
+    ///
+    /// # Returns
+    ///
+    /// * `AccountTransaction` - The converted transaction
+    pub fn to_declare_tx(&self) -> DeclareTransaction {
```

Comment:
```suggestion
    pub fn try_into_declare_tx(&self) -> DeclareTransaction {
```
return result and remove all unwraps